### PR TITLE
HOUSNAV-22: question footer and initial walkthrough state stores

### PR DESCRIPTION
**Pull request by nicholaslyon-dt**
[HOUSNAV-22](https://hous-bssb.atlassian.net/browse/HOUSNAV-22)

## Overview & Purpose
These changes add a footer that looks like the design and improves the overall walkthrough display to be close to the design.

## Summary of Changes
Added a WalkthroughStore
Updated footer, walkthrough, and question components to better match the design
Updated walkthrough and child components to consume the WalkthroughStore
Added the tertiary button type for the back button in the footer
Added icons for the buttons in the footer

## Side Effects
The includes a complete refactoring of how the walkthrough was handling state. It now happens in a central store. This store is added to a react Provider at the top level of the walkthrough. Each component in the walkthrough can now observe the store and react to changes as needed on it's own.

## Testing
1. Navigate to the walkthrough
2. Observe the disabled buttons in the footer to match the design
3. Click an answer and observe the next button being enabled
4. Click next and see question #2, observe the next button is disabled again
5. Observe the back button is now enabled.
6. Click back button and observe question #1 again with previous answer shown.
7. Click back and forth observing state change of footer and questions
8. Observe question scrolling back to the top if next/back is clicked while scrolled father down the question

## Notes & Resources
There are only 3 possible questions. Click next on question #3 just goes back to question #2.

### CODE REVIEW

**Comment by jbook24** [comment]
Looks good! Curious do we feel this could benefit from additional margin? 
**Current** 
image


**Proposed** 
image

File: apps/web/components/walkthrough/Walkthrough.css [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    grid-template-columns: var(--walkthrough-side-nav-width) 1fr;
+  }
+
+  .web-Walkthrough--SideNav {
+    display: block;
+  }
+}
+
+/* DESKTOP - 912px */
+@media (min-width: 57rem) {
```
Comment (jbook24) [inline apps/web/components/walkthrough/Walkthrough.css]: Should we explore adding a responsive.css file that houses media-queries definitions and then we can reference those without needing to annotate?

**Comment by laurenrevere** [comment]
> Looks good! Curious do we feel this could benefit from additional margin? **Current** image
> 
> **Proposed** image

Here's the ticket for that! https://hous-bssb.atlassian.net/browse/HOUSNAV-52

The current implementation matches the designs, but we asked Zach if he was okay with us adding some space.

File: apps/web/package.json [hunk 2]
```diff
@@ -21,6 +21,8 @@
     "axe-core": "^4.9.1",
     "cypress-axe": "^1.5.0",
     "html-react-parser": "^5.1.10",
+    "mobx": "^6.12.3",
+    "mobx-react-lite": "^4.0.7",
```
Comment (jbook24) [inline apps/web/package.json:25]: This repo doesn't look like it's being maintained. Should we explore an alternative? 
https://github.com/mobxjs/mobx-react-lite

File: apps/web/stores/WalkthroughRootStore.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      return undefined;
+
+    return currentQuestion;
+  }
+
+  setCurrentQuestionId = (questionId: string) => {
+    this.currentQuestionId = questionId;
+  };
+
+  // get currentSectionId() {
```
Comment (jbook24) [inline apps/web/stores/WalkthroughRootStore.ts]: Should this be removed?

File: apps/web/components/walkthrough/Walkthrough.css (hunk 1 above)
Comment (nicholaslyon-dt) [reply apps/web/components/walkthrough/Walkthrough.css]: Sadly we can't. The one drawback of vanilla CSS currently is the inability to use custom properties in media queries. The ONE thing I don't like about this setup. This is why I've been trying to keep the media queries to the variables.css file as much as possible with just updating the custom properties there. I just committed a refactoring that does this for a few cases here.

File: apps/web/package.json (hunk 2 above)
Comment (nicholaslyon-dt) [reply apps/web/package.json:25]: Yeah that's the old package. The one we're getting comes from here:

https://github.com/mobxjs/mobx/tree/main/packages/mobx-react-lite

They moved it to a package of the main mobx.
https://github.com/mobxjs/mobx-react-lite#-this-repo-has-been-moved-to-mobx

Currently 100 on Snyk!
https://snyk.io/advisor/npm-package/mobx

File: apps/web/stores/WalkthroughRootStore.ts (hunk 3 above)
Comment (nicholaslyon-dt) [reply apps/web/stores/WalkthroughRootStore.ts]: Good call. Removed!

**Comment by laurenrevere** [comment]
On safari on mobile, the footer is hidden under the address bar. This may lead to users not knowing how to proceed. Might we want to [add padding on the bottom -- maybe only on safari](https://stackoverflow.com/questions/31541998/ios-mobile-safari-the-bottom-bar-covers-my-footer)?

image

**Comment by nicholaslyon-dt** [comment]
> On safari on mobile, the footer is hidden under the address bar. This may lead to users not knowing how to proceed. Might we want to [add padding on the bottom -- maybe only on safari](https://stackoverflow.com/questions/31541998/ios-mobile-safari-the-bottom-bar-covers-my-footer)?
> 
> image

Man I forgot about this shit from Apple. I hate mobile Safari.

**Comment by nicholaslyon-dt** [comment]
@laurenrevere NVM, much easier to fix now! Dynamic Viewport Height FTW. Can you verify this works on your end? Worked in BrowserStack for me.

**Comment by laurenrevere** [comment]
> @laurenrevere NVM, much easier to fix now! Dynamic Viewport Height FTW. Can you verify this works on your end? Worked in BrowserStack for me.

Looks good now!

**Review by jbook24 — APPROVED** [review]

**Comment by laurenrevere** [comment]
When you tab through a page that needs to scroll, the footer resizes to fill most of the page. Tested on both chrome and safari.

Screenshot 2024-06-10 at 9 41 26 AM

File: apps/web/components/question/question-types/QuestionMissing.tsx [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export default function QuestionMissing() {
+  // TODO - better error screen
```
Comment (wardm89) [inline apps/web/components/question/question-types/QuestionMissing.tsx:2]: Is there a ticket for this?

Comment (laurenrevere) [reply apps/web/components/question/question-types/QuestionMissing.tsx:2]: We don't currently have any designs for them. I also don't see a ticket for it.

Comment (laurenrevere) [reply apps/web/components/question/question-types/QuestionMissing.tsx:2]: Created a low priority design ticket for this: https://hous-bssb.atlassian.net/browse/HOUSNAV-70

**Pull request merged**